libraries/utilities/generic.cpp

```diff
@@ -0,0 +1,16 @@
+#include <string>
```

Comment:
These methods should not be defined or used, where ever they are used we need to do something different.  I do not even se their use in this commit.